inc/Engine/CriticalPath/CriticalCSSSubscriber.php

```diff
@@ -688,6 +689,25 @@ public function maybe_regenerate_cpcss() {
 		$this->critical_css->process_handler();
 	}
 
+	/**
+	 * Maybe disable adding the preload fonts links
+	 *
+	 * @since 3.8.8
+	 *
+	 * @return bool
+	 */
+	public function maybe_disable_preload_fonts() : bool {
+		if ( ! $this->should_async_css() ) {
+			return false;
+		}
+
+		return (
+			! empty( $this->critical_css->get_current_page_critical_css() )
+			||
+			! empty( $this->options->get( 'critical_css', '' ) )
+		);
```

Comment:
```suggestion
		return (
			$this->should_async_css()
			&&
			(
				! empty( $this->options->get( 'critical_css', '' ) )
				||
				! empty( $this->critical_css->get_current_page_critical_css() )
			)
		);
```

what about this small change?
**Why?**
- Check the database option before checking the filesystem.
- **(Optional)** merge `should_async_css` into the check. (maybe you did such line to return earlier or to make it more readable, just commented here to learn from u)

Reply:
Good idea to check for the database first, it's likely a bit faster.

In my opinion, keeping the first check on its own makes the code more readable, when you start to accumulate multiple conditions and operators, it becomes hard to see the path.

Reply:
Got it, thanks.